custom_components/tahoma/alarm_entities/tsk_alarm_controller.py

```diff
@@ -0,0 +1,77 @@
+"""Support for Overkiz TSKAlarmController."""
+from __future__ import annotations
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.alarm_control_panel import AlarmControlPanelEntity
+from homeassistant.components.alarm_control_panel.const import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_ARM_NIGHT,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_PENDING,
+    STATE_ALARM_TRIGGERED,
+)
+
+from ..entity import OverkizEntity
+
+
+class TSKAlarmController(OverkizEntity, AlarmControlPanelEntity):
+    """Representation of a Overkiz Alarm Control Panel."""
+
+    _attr_supported_features = (
+        SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_NIGHT
+    )
+    _attr_entity_registry_enabled_default = False
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if (
+            self.executor.select_state(OverkizState.INTERNAL_INTRUSION_DETECTED)
+            == OverkizCommandParam.DETECTED
+        ):
+            return STATE_ALARM_TRIGGERED
+
+        if self.executor.select_state(
+            OverkizState.INTERNAL_CURRENT_ALARM_MODE
+        ) != self.executor.select_state(OverkizState.INTERNAL_TARGET_ALARM_MODE):
+            return STATE_ALARM_PENDING
+
+        MAP_INTERNAL_STATUS_STATE = {
+            OverkizCommandParam.OFF: STATE_ALARM_DISARMED,
+            "zone1": STATE_ALARM_ARMED_HOME,
+            "zone2": STATE_ALARM_ARMED_NIGHT,
+            OverkizCommandParam.TOTAL: STATE_ALARM_ARMED_AWAY,
```

Comment:
I will need to add `zone1` and `zone2` to pyoverkiz. This is not according to Somfy spec. It should be `partial1` and `partial2`, but in my and other users tests it seems that they use zone instead of partial.